internal/command/test.go

```diff
@@ -731,7 +734,10 @@ func (runner *TestFileRunner) plan(config *configs.Config, state *states.State,
 	references, referenceDiags := run.GetReferences()
 	diags = diags.Append(referenceDiags)
 
-	variables, variableDiags := buildInputVariablesForTest(run, file, config, runner.Suite.GlobalVariables, runner.States)
+	evalCtx, ctxDiags := getEvalContextForTest(runner.States, config, runner.Suite.GlobalVariables)
```

Comment:
Just making sure - I saw you added this call to the `runner.plan` and `runner.destory`, and I was wondering if we also need it in the `runner.apply`? 
Or maybe we don't need it because in the `runner.apply` stage we are relying on the plan that was created using the `runner.plan`, and the variables are already declared as part of the plan.

Reply:
Yes, correct. In the case of `runner.apply`, we are using variables from the [plan](https://github.com/opentofu/opentofu/blob/7a713ccd833c10a315e2397da80923215da45332/internal/command/test.go#L504) that is part of [runner.plan](https://github.com/opentofu/opentofu/blob/7a713ccd833c10a315e2397da80923215da45332/internal/command/test.go#L723).